File: c#/classes/ClassComponents.cs
class Account {
  public string Login = "";
  public string Email = "";
  public bool IsActive = false;
  public int Id = 0;

  public int[] Connections;
  public int ConnectionsNumber = 0;

  public void Display() {
    System.Console.WriteLine("ID: " + Id + "\n"
      + "Login: " + Login + "\n"
      + "E-mail: " + Email + "\n"
      + "Is active: " + IsActive + "\n"
      + "Has connections: " + HasConnections()
    );
  }

  bool HasConnections() {
    return (ConnectionsNumber > 0);
  }
}

class ClassComponents {
  public static void Main(string[] args) {
    Account timothy = new Account();

    timothy.Login = "tim";
    timothy.Email = "timothy.muppetone@gmail.com";
    timothy.IsActive = true;

    timothy.Display();
  }
}

File: perl/classes/class_components.pl
#!/usr/bin/perl

package Account;

use strict;
use warnings;

sub new {
  my ($class, $args) = @_;
  my $self = bless {
    login => "",
    email => "",
    is_active => 0,
    id => 0,

    connections => [],
    connections_number => 0,
  }, $class;
}

sub display {
  my $self = shift;

  print("ID: " . $self->{id} . "\n"
    . "Login: " . $self->{login} . "\n"
    . "E-mail: " . $self->{email} . "\n"
    . "Is active: " . $self->{is_active} . "\n"
    . "Has connections: " . $self->has_connections() . "\n"
  );
};

sub has_connections {
  my $self = shift;

  return ($self->{connections_number} > 0) ? 1 : 0;
};

my $timothy = Account->new();

$timothy->{login} = "tim";
$timothy->{email} = "timothy.muppetone\@gmail.com";
$timothy->{is_active} = 1;

$timothy->display();
